tools/matc/src/matc/ParametersProcessor.cpp

```diff
@@ -361,201 +251,261 @@ bool ParametersProcessor::processVariables(filamat::MaterialBuilder& builder,
+        { "lit", MaterialBuilder::Shading::LIT },
+        { "subsurface", MaterialBuilder::Shading::SUBSURFACE },
+        { "unlit", MaterialBuilder::Shading::UNLIT },
+        { "specularGlossiness", MaterialBuilder::Shading::SPECULAR_GLOSSINESS },
+    };
     auto jsonString = value.toJsonString();
-    if (!isStringValidEnum(mStringToShading, jsonString->getString())) {
-        return logEnumIssue(PARAM_KEY_SHADING, *jsonString, mStringToShading);
+    if (!isStringValidEnum(strToEnum, jsonString->getString())) {
+        return logEnumIssue("shading", *jsonString, strToEnum);
     }
 
-    builder.shading(stringToEnum(mStringToShading, jsonString->getString()));
+    builder.shading(stringToEnum(strToEnum, jsonString->getString()));
     return true;
 }
 
-bool ParametersProcessor::processVariantFilter(filamat::MaterialBuilder& builder,
-        const JsonishValue& value) {
+static bool processVariantFilter(MaterialBuilder& builder, const JsonishValue& value) {
+    static const std::unordered_map<std::string, uint8_t> strToEnum {
+        { "directionalLighting", filament::Variant::DIRECTIONAL_LIGHTING },
+        { "dynamicLighting", filament::Variant::DYNAMIC_LIGHTING },
+        { "shadowReceiver", filament::Variant::SHADOW_RECEIVER },
+        { "skinning", filament::Variant::SKINNING },
+    };
     uint8_t variantFilter = 0;
     const JsonishArray* jsonArray = value.toJsonArray();
     const auto& elements = jsonArray->getElements();
 
     for (size_t i = 0; i < elements.size(); i++) {
         auto elementValue = elements[i];
         if (elementValue->getType() != JsonishValue::Type::STRING) {
-            std::cerr << PARAM_KEY_VARIANT_FILTER << ": array index " << i <<
+            std::cerr << "variant_filter: array index " << i <<
                       " is not a STRING. found:" <<
                       JsonishValue::typeToString(elementValue->getType()) << std::endl;
             return false;
         }
 
         const std::string& s = elementValue->toJsonString()->getString();
-        if (!isStringValidEnum(mStringToVariant, s)) {
-            std::cerr << PARAM_KEY_VARIANT_FILTER << ": variant " << s <<
+        if (!isStringValidEnum(strToEnum, s)) {
+            std::cerr << "variant_filter: variant " << s <<
                       " is not a valid variant" << std::endl;
         }
 
-        variantFilter |= mStringToVariant[s];
+        variantFilter |= strToEnum.at(s);
     }
 
     builder.variantFilter(variantFilter);
     return true;
 }
 
-filamat::MaterialBuilder::Variable ParametersProcessor::intToVariable(size_t i) const noexcept {
-    switch (i) {
-        case 0:  return MaterialBuilder::Variable::CUSTOM0;
-        case 1:  return MaterialBuilder::Variable::CUSTOM1;
-        case 2:  return MaterialBuilder::Variable::CUSTOM2;
-        case 3:  return MaterialBuilder::Variable::CUSTOM3;
-        default: return MaterialBuilder::Variable::CUSTOM0;
+ParametersProcessor::ParametersProcessor() {
```

Comment:
Can we move this to the top so it's easier to find that list of parameters when we add new ones?

Reply:
Sure but I'll also need to replace the static funcs with lambdas, otherwise we'd have to forward-declare each function (which doesn't improve the existing maintenance burden).  I'll add another commit to the PR for this.

Reply:
Ah good point. It's fine as is.